storage/database/db_manager.go

```diff
@@ -44,192 +44,194 @@ var (
-	ReadTxHashFromSenderTxHash(senderTxHash common.Hash) common.Hash
-
-	ReadBloomBits(bloomBitsKey []byte) ([]byte, error)
-	WriteBloomBits(bloomBitsKey []byte, bits []byte) error
-
-	ReadValidSections() ([]byte, error)
-	WriteValidSections(encodedSections []byte)
-
-	ReadSectionHead(encodedSection []byte) ([]byte, error)
-	WriteSectionHead(encodedSection []byte, hash common.Hash)
-	DeleteSectionHead(encodedSection []byte)
-
-	// from accessors_metadata.go
-	ReadDatabaseVersion() *uint64
-	WriteDatabaseVersion(version uint64)
-
-	ReadChainConfig(hash common.Hash) *params.ChainConfig
-	WriteChainConfig(hash common.Hash, cfg *params.ChainConfig)
-
-	// below operations are used in parent chain side, not child chain side.
-	WriteChildChainTxHash(ccBlockHash common.Hash, ccTxHash common.Hash)
-	ConvertChildChainBlockHashToParentChainTxHash(scBlockHash common.Hash) common.Hash
-
-	WriteLastIndexedBlockNumber(blockNum uint64)
-	GetLastIndexedBlockNumber() uint64
-
-	// below operations are used in child chain side, not parent chain side.
-	WriteAnchoredBlockNumber(blockNum uint64)
-	ReadAnchoredBlockNumber() uint64
-
-	WriteReceiptFromParentChain(blockHash common.Hash, receipt *types.Receipt)
-	ReadReceiptFromParentChain(blockHash common.Hash) *types.Receipt
-
-	WriteHandleTxHashFromRequestTxHash(rTx, hTx common.Hash)
-	ReadHandleTxHashFromRequestTxHash(rTx common.Hash) common.Hash
-
-	WriteParentOperatorFeePayer(feePayer common.Address)
-	WriteChildOperatorFeePayer(feePayer common.Address)
-	ReadParentOperatorFeePayer() common.Address
-	ReadChildOperatorFeePayer() common.Address
-
-	// cacheManager related functions.
-	ClearHeaderChainCache()
-	ClearBlockChainCache()
-	ReadTxAndLookupInfoInCache(hash common.Hash) (*types.Transaction, common.Hash, uint64, uint64)
-	ReadBlockReceiptsInCache(blockHash common.Hash) types.Receipts
-	ReadTxReceiptInCache(txHash common.Hash) *types.Receipt
-
-	// snapshot in clique(ConsensusClique) consensus
-	WriteCliqueSnapshot(snapshotBlockHash common.Hash, encodedSnapshot []byte) error
-	ReadCliqueSnapshot(snapshotBlockHash common.Hash) ([]byte, error)
-
-	// Governance related functions
-	WriteGovernance(data map[string]interface{}, num uint64) error
-	WriteGovernanceIdx(num uint64) error
-	ReadGovernance(num uint64) (map[string]interface{}, error)
-	ReadRecentGovernanceIdx(count int) ([]uint64, error)
-	ReadGovernanceAtNumber(num uint64, epoch uint64) (uint64, map[string]interface{}, error)
-	WriteGovernanceState(b []byte) error
-	ReadGovernanceState() ([]byte, error)
-
-	// StakingInfo related functions
-	ReadStakingInfo(blockNum uint64) ([]byte, error)
-	WriteStakingInfo(blockNum uint64, stakingInfo []byte) error
-
-	// DB migration related function
-	StartDBMigration(DBManager) error
-
-	// ChainDataFetcher checkpoint function
-	WriteChainDataFetcherCheckpoint(checkpoint uint64) error
-	ReadChainDataFetcherCheckpoint() (uint64, error)
-}
+type (
+	DBManager interface {
```

Comment:
What's the purpose of nesting type declaration here?

Reply:
There is no purpose. go fmt changed this.

Reply:
I think I made a mistake and go fmat changed this.
I will revert. :)